### ISSUE TITLE
`IsExternalInit` not generated correctly in multitargeting projects

### ISSUE THREAD

**Issue by Lordfirespeed** [Sergio0694/PolySharp#98]
### Reproduction Steps

Make a project that targets `netstandard2.1` and `net8.0`

```xml
<TargetFrameworks>netstandard2.1;net8.0</TargetFrameworks>
```

and add a `record class` to it with some members, or alternatively a class with `required init` properties.

### Expected Behavior

`IsExternalInits` type should be included in the compiled assembly targeting `netstandard2.1`.

### Actual Behavior

Compilation succeeds, but `IsExternalInits` type is not included in the compiled assembly targeting `netstandard2.1`.

### System info

This section should contain useful info such as:
- **PolySharp** NuGet version 1.14..1
- Operating system version Pop!_OS (Ubuntu) 22.04
- .NET SDK version 8.0.100
- Runtime is Unity 2022.3.9f1 (implements `net48` and `netstandard2.1`)

**Comment by AArnott** [comment]
`IsExternalInit` _must_ be emitted all the time, for all frameworks. Even if the framework already declares it. 
The explanation of why is here: https://github.com/manuelroemer/IsExternalInit/issues/5

**Comment by Sergio0694** [comment]
> "`IsExternalInit` must be emitted all the time, for all frameworks."

This is not accurate. PolySharp generates the correct `[TypeForwardedTo]` in that case, instead of the polyfill. Not generating the polyfill is only a binary breaking changes if you _don't_ generate the correct forwards, but PolySharp does that, so that's fine.

> "Compilation succeeds, but `IsExternalInits` type is not included in the compiled assembly targeting netstandard2.1."

@Lordfirespeed I need more info from your end. First, I can't repro this:

![image](https://github.com/Sergio0694/PolySharp/assets/10199417/e4de4fe9-5307-4f24-bd0b-db436b833b05)

Second, how is your project even compiling at all, if you're saying the .NET Standard 2.1 target doesn't have that polyfill?

Are you sure you're not looking at the wrong place or perhaps it was just VS not showing all generated files correctly?

**Comment by AArnott** [comment]
@Sergio0694 You are quite right. Brilliant solution that I hadn't thought of so didn't check for. My apologies for spreading FUD.

**Comment by Sergio0694** [comment]
All good, no worries! Just wanted to make sure there were no misunderstandings! 😊

**Comment by teo-tsirpanis** [comment]
Cannot reproduce with PolySharp 1.15.0 and .NET SDK 9.0.201.
